### DISCUSSION TITLE
مشکل در بازگشت از زرین پال

### DISCUSSION THREAD

**Discussion by hamedkermani62** [Sina-Soltani/Parbad#337] [Ideas]
سلام
ما در .netcore از parbad استفاده کردیم. روی لوکال هم درسته. هم sandbox و هم از لوکال وارد بانک میشه و برمیگرده و پرداخت انجام میشه.
اما وقتی روی سرور قرار میگیره، بعد از برگشت از بانک و کم کردن مبلغ، وقتی وارد صفحه بازگشت خودمون میشه خطا میده و نمیتونه token number رو fetch کنه.
url  هم درسته و توکن نامبر توش هست.
میتونید راهنمایی کنید ؟

**Comment by Sina-Soltani** [comment]
درود
کدهای قسمت تنظیمات و قسمت پرداخت رو قرار بدید تا بررسی کنیم.

**Comment by hamedkermani62** [comment]
صفحه برگشت از بانک 404 میده رو سرور
اما روی لوکال کار میکنه

        [Route("Verify")]
        public async Task<IActionResult> Verify()
        {
           
            var invoice = await _onlinePayment.FetchAsync();
            var order = _orderService.GetOrderByTrackingNumber(invoice.TrackingNumber);


            if (invoice.Status == PaymentFetchResultStatus.AlreadyProcessed)
            {
                var isAlreadyVerified = invoice.IsAlreadyVerified;
                return View("پرداخت قبلاً پردازش شده است.");
            }

            var verifyResult = await _onlinePayment.VerifyAsync(invoice);

            if (verifyResult.IsSucceed)
            {
                var transactionCode = verifyResult.TransactionCode;
                ViewBag.IsSuccess = true;
                order.IsFinaly = true;
                order.TransactionCode = transactionCode;
                order.OrderStatus = 1;
                _orderService.UpdateOrder(order);

                return View(verifyResult);
            }
            else
            {
                var cancelResult = await _onlinePayment.CancelAsync(invoice, cancellationReason: "تراکنش لغو شد و مبلغ بازگشت داده شد");
                //return View("CancelResult", cancelResult);
                return View(verifyResult);
            }


        }






ارسال به درگاه 

    var result = await _onlinePayment.RequestAsync(invoice =>
                {
                    invoice
                        .SetAmount(Convert.ToDecimal(price))
                        .SetCallbackUrl(verifyUrl)
                        .SetZarinPalData($"پرداخت فاکتور شماره {order.OrderId}", "", "")
                        .SetTrackingNumber(order.OrderId)
                        .SetGateway("ZarinPal");

                    invoice.UseAutoRandomTrackingNumber();
                });


                if (result.IsSucceed)
                {
                    _orderService.AddOrderAddresse(order.OrderId, AddresseId);
                    order.TrackingNumber = result.TrackingNumber;
                    _orderService.UpdateOrder(order);
                    return result.GatewayTransporter.TransportToGateway();
                }
                else
                {
                    return BadRequest();
                }




satartup
      services.AddParbad()
                 .ConfigureGateways(gateways =>
                 {

                     gateways
                          .AddZarinPal()
                          .WithAccounts(accounts =>
                          {
                              accounts.AddInMemory(account =>
                              {
                                  account.MerchantId = "000000000";
                                  account.IsSandbox = false;
                              });
                          });
                 })
                 .ConfigureHttpContext(builder => builder.UseDefaultAspNetCore())
                 .ConfigureStorage(builder => builder.UseMemoryCache());

**Comment by Sina-Soltani** [comment]
کدها که درست هست و مشکلی من نمیبینم.
متاسفانه کمکی از دست من بر نمیاد، چون این مشکل ظاهرا داخلی و مربوط به سیستم شما هست.
